app/src/main/java/com/globant/pokemontcgapp/fragment/PokemonTypeFragment.kt

```diff
@@ -32,20 +34,40 @@ class PokemonTypeFragment : Fragment() {
         pokemonTypeViewModel.getPokemonTypes()
     }
 
-    private fun updateUI(pokemonTypes: PokemonTypeData) {
-        when (pokemonTypes.state) {
-            PokemonTypeState.LOADING -> binding.pokemonTypeProgressBar.visibility = View.VISIBLE
-            PokemonTypeState.POKEMON_TYPE_DATA -> pokemonTypes.data?.pokemonTypesList?.let { showPokemonTypes(it) }
+    override fun onPause() {
+        binding.pokemonTypeRecyclerView.visibility = View.GONE
+        super.onPause()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        pokemonTypeViewModel.getPokemonTypes()
+    }
+
+    private fun updateUI(data: Event<Data<List<PokemonType>>>) {
+        val pokemonTypesData = data.getContentIfNotHandled()
+        when (pokemonTypesData?.status) {
+            Status.LOADING -> binding.pokemonTypeProgressBar.visibility = View.VISIBLE
+            Status.SUCCESS -> pokemonTypesData.data?.let { showPokemonTypes(it) }
+            Status.ERROR -> showPokemonTypesError(pokemonTypesData.error?.message)
         }
     }
 
-    private fun showPokemonTypes(pokemonTypes: List<String>) {
-        pokemonTypesAdapter.submitList(pokemonTypes)
+    private fun showPokemonTypes(pokemonTypes: List<PokemonType>) {
+        val pokemonTypesAdapter = PokemonTypesAdapter(pokemonTypes)
+
+        binding.pokemonTypeProgressBar.visibility = View.GONE
+        binding.pokemonTypeRecyclerView.apply {
+            layoutManager =
+                GridLayoutManager(context, resources.configuration.getColumnsByOrientation(COLUMNS_PORTRAIT, COLUMNS_LANDSCAPE))
+            adapter = pokemonTypesAdapter
```

Comment:
u can instantiate the adapter here, and avoid var

Reply:
refactored, thanks